src/transformers/models/tvp/modeling_tvp.py

```diff
@@ -0,0 +1,1199 @@
+            stride=stride,
+            padding=1,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(out_channels),
+        )
+
+        self.conv2 = TVPVisionConv2d(
+            out_channels,
+            out_channels,
+            kernel_size=3,
+            stride=1,
+            padding=1,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(out_channels),
+        )
+
+        for layer in [self.conv1, self.conv2, self.shortcut]:
+            if layer is not None:  # shortcut can be None
+                c2_msra_fill(layer)
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = F.relu_(out)
+        out = self.conv2(out)
+
+        if self.shortcut is not None:
+            shortcut = self.shortcut(x)
+        else:
+            shortcut = x
+
+        out += shortcut
+        out = F.relu_(out)
+        return out
+
+
+class TVPVisionBottleneckBlock(TVPVisionBlockBase):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        bottleneck_channels,
+        stride=1,
+        num_groups=1,
+        dilation=1,
+    ):
+        super().__init__(in_channels, out_channels, stride)
+
+        if in_channels != out_channels:
+            self.shortcut = TVPVisionConv2d(
+                in_channels,
+                out_channels,
+                kernel_size=1,
+                stride=stride,
+                bias=False,
+                norm=TVPVisionFrozenBatchNorm2d(out_channels),
+            )
+        else:
+            self.shortcut = None
+
+        self.conv1 = TVPVisionConv2d(
+            in_channels,
+            bottleneck_channels,
+            kernel_size=1,
+            stride=stride,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(bottleneck_channels),
+        )
+
+        self.conv2 = TVPVisionConv2d(
+            bottleneck_channels,
+            bottleneck_channels,
+            kernel_size=3,
+            stride=1,
+            padding=dilation,
+            bias=False,
+            groups=num_groups,
+            dilation=dilation,
+            norm=TVPVisionFrozenBatchNorm2d(bottleneck_channels),
+        )
+
+        self.conv3 = TVPVisionConv2d(
+            bottleneck_channels,
+            out_channels,
+            kernel_size=1,
+            bias=False,
+            norm=TVPVisionFrozenBatchNorm2d(out_channels),
+        )
+
+        for layer in [self.conv1, self.conv2, self.conv3, self.shortcut]:
+            if layer is not None:  # shortcut can be None
+                c2_msra_fill(layer)
+
+    def forward(self, x):
+        out = self.conv1(x)
+        out = F.relu_(out)
+
+        out = self.conv2(out)
+        out = F.relu_(out)
+
+        out = self.conv3(out)
+
+        if self.shortcut is not None:
+            shortcut = self.shortcut(x)
+        else:
+            shortcut = x
+
+        out += shortcut
+        out = F.relu_(out)
+        return out
+
+
+class TVPVisionBackbone(nn.Module):
```

Comment:
Is this backbone a standard resnet? In this case it should be loaded using our AutoBackbone API. There's a [ResNetBackbone](https://github.com/huggingface/transformers/blob/f6295c6c535c2b036a4533327ab5a92c6b199b78/src/transformers/models/resnet/modeling_resnet.py#L432) already implemented

Reply:
Hi @amyeroberts .

I am willing to use [ResNetBackbone](https://github.com/huggingface/transformers/blob/f6295c6c535c2b036a4533327ab5a92c6b199b78/src/transformers/models/resnet/modeling_resnet.py#L432) because it is more convenient. However, the backbone of the original pre-trained TVP checkpoint is slightly different from `ResNetBackbone`.
See the following images:
TVP backbone:
![65c04de8-88a2-41cb-ac42-93b43c79c2ea](https://github.com/huggingface/transformers/assets/107918818/ffaef90c-3307-4fb5-b1df-38f936bdf586)

ResNet backbone
![b5c24ec8-d7e7-4bf6-b5fd-0ba2f6416573](https://github.com/huggingface/transformers/assets/107918818/3691fd8e-0a89-4902-9a2b-60625b90e7a1)

We can see that the `stride` is different. Moreover, the [ResNetBackbone](https://github.com/huggingface/transformers/blob/main/src/transformers/models/resnet/modeling_resnet.py#L226) did not support control `stride` in `ResNetEncoder`. 

There is a possible way, we can change `ResNet` model to support manipulating `stride` in the ResNet backbone. Otherwise, we can only keep TVP backbone in our source code.

Would like to hear your opinion. Thx!

Reply:
You can probaly use:
```python 
class ResNetEncoder(nn.Module):
    def __init__(self, config: ResNetConfig):
        super().__init__()
        self.stages = nn.ModuleList([])
        # based on `downsample_in_first_stage` the first layer of the first stage may or may not downsample the input
        self.stages.append(
            ResNetStage(
                config,
                config.embedding_size,
                config.hidden_sizes[0],
                stride=2 if config.downsample_in_first_stage else 1,
                depth=config.depths[0],
            )
        )
        in_out_channels = zip(config.hidden_sizes, config.hidden_sizes[1:])
        for (in_channels, out_channels), depth in zip(in_out_channels, config.depths[1:]):
            self.stages.append(ResNetStage(config, in_channels, out_channels, depth=depth))
```
so setting `config.downsample_in_first_stage` to the appropriate value no?

Reply:
Hi @ArthurZucker . Thanks for your advice, but the differences still remain even if I set `downsample_in_first_stage=True` in `config.json`.